src/web/assets/cp/src/js/CustomizeSourcesModal.js

```diff
@@ -129,262 +178,307 @@ Craft.CustomizeSourcesModal = Garnish.Modal.extend({
+    // Select this by default?
+    if (
+      this.elementIndex.settings.page &&
+      Craft.CustomizeSourcesModal.Page.nameId(
+        this.elementIndex.settings.page
+      ) === Craft.CustomizeSourcesModal.Page.nameId(name)
+    ) {
+      page.select();
+    }
 
-    this.closeSidebar();
+    this.pages.push(page);
+    this.updatePageActionButtons();
 
-    // Add listeners
-    this.addListener(this.$sidebarToggleBtn, 'click', () => {
-      this.toggleSidebar();
-    });
+    if (isNew) {
+      Craft.cp.announce(Craft.t('app', 'Success'));
+    }
 
-    this.addListener(this.$sidebarCloseBtn, 'click', () => {
-      this.toggleSidebar();
-      this.$sidebarToggleBtn.focus();
-    });
+    return page;
   },
 
-  toggleSidebar: function () {
-    if (this.sidebarIsOpen()) {
-      this.closeSidebar();
-    } else {
-      this.openSidebar();
-    }
+  focusLabelInput: function () {
+    this.selectedSource.$labelInput.focus();
   },
 
-  openSidebar: function () {
-    this.$container.removeClass('sidebar-hidden');
-    this.$sidebarToggleBtn.attr('aria-expanded', 'true');
-    this.$sidebar.find(':focusable').first().focus();
-
-    Garnish.uiLayerManager.addLayer(this.$sidebar);
-
-    Garnish.uiLayerManager.registerShortcut(Garnish.ESC_KEY, () => {
-      this.closeSidebar();
-
-      if (Garnish.focusIsInside(this.$sidebar)) {
-        this.$sidebarToggleBtn.focus();
-      }
-    });
+  getSourceName: function () {
+    return this.selectedSource
+      ? this.selectedSource.sourceData.label
+      : this.sources[0].sourceData.label;
   },
 
-  closeSidebar: function () {
-    this.$container.addClass('sidebar-hidden');
-
-    if (this.$sidebarToggleBtn) {
-      this.$sidebarToggleBtn.attr('aria-expanded', 'false');
-    }
-
-    // if sidebar is topmost layer, remove layer
-    if (Garnish.uiLayerManager.currentLayer.$container.hasClass('cs-sidebar')) {
-      Garnish.uiLayerManager.removeLayer();
+  getSourceContainer: function (pageName, create = true) {
+    if (this.sourceContainers[pageName] === undefined && create) {
+      this.sourceContainers[pageName] = $('<ol class="cs-sidebar-list">');
+      if (this.$newSourceBtn) {
+        this.sourceContainers[pageName].insertBefore(this.$newSourceBtn);
+      } else {
+        this.sourceContainers[pageName].appendTo(this.$sourcesSidebarContent);
+      }
+      if (this.multiPage && pageName !== this.selectedPage.name) {
+        this.sourceContainers[pageName].addClass('hidden');
+      }
     }
-  },
 
-  sidebarIsOpen: function () {
-    return this.$sidebarToggleBtn.attr('aria-expanded') === 'true';
+    return this.sourceContainers[pageName];
   },
 
   addSource: function (sourceData, isNew) {
-    const $item = $('<div class="customize-sources-item"/>').appendTo(
-      this.$sourcesContainer
-    );
-    const $itemButton = $('<div class="customize-sources-item__btn"/>')
+    const pageName = sourceData.page ?? this.selectedPage.name ?? '__DEFAULT__';
```

Comment:
Missing a check for `this.multiPage`? Modal blows up on categories index, when trying to access `selectedPage`. 😉